batch/batch/server/database.py

```diff
@@ -135,6 +136,49 @@ def __init__(self, db):
     async def get_records_where(self, condition):
         return await super().get_records(condition)
 
+    async def find_records(self, user, complete=None, success=None, deleted=None, attributes=None):
+        sql = "select batch.* from {self.name} as batch"
+        values = []
+        joins = []
+        wheres = []
+        havings = []
+        groups = []
+
+        values += user
+        wheres += "job.user = %s"
+        if deleted is not None:
+            if deleted:
+                wheres += "batch.deleted"
+            else:
+                wheres += "not batch.deleted"
+        if complete is not None:
+            condition = "not batch.is_open and batch.n_completed = batch.n_jobs"
+            if complete:
+                wheres += condition
+            else:
+                wheres += f"not ({condition})"
+        if success is not None:
+            condition = "not batch.is_open and batch.n_succeeded = batch.n_jobs"
+            if success:
+                wheres += condition
+            else:
+                wheres += f"not ({condition})"
+        if attributes:
+            joins += "inner join {self._db.batch_attributes.name} as attr using (batch_id)"
+            groups += "batch.id"
+            for k, v in attributes.items():
+                values += v
+                values += len(attributes)
+                havings += f"sum(attr.`{k}` = %s) = %s"
```

Comment:
I don't see how this line is correct. Shouldn't the second `%s` be 1 and not `len(attributes)`?